> The CPU version build fine, but CUDA fails with:
> 
> ```
> CMake Error at /nix/store/aqckch626lg0vxh41dabyzrq0jx7gdk5-cmake-3.29.6/share/cmake-3.29/Modules/FindCUDA/select_compute_arch.cmake:245 (message):
>   Unknown CUDA Architecture Name 9.0a in CUDA_SELECT_NVCC_ARCH_FLAGS
> Call Stack (most recent call first):
>   cmake/public/utils.cmake:398 (cuda_select_nvcc_arch_flags)
>   torch/CMakeLists.txt:338 (torch_cuda_get_nvcc_gencode_flag)
> 
> 
> CMake Error at /nix/store/aqckch626lg0vxh41dabyzrq0jx7gdk5-cmake-3.29.6/share/cmake-3.29/Modules/FindCUDA/select_compute_arch.cmake:249 (message):
>   arch_bin wasn't set for some reason
> Call Stack (most recent call first):
>   cmake/public/utils.cmake:398 (cuda_select_nvcc_arch_flags)
>   torch/CMakeLists.txt:338 (torch_cuda_get_nvcc_gencode_flag)
> ```

Ok, the issue is that `select_compute_arch.cmake` is removed and then the counterpart in CMake is picked up, which isn't patched to recognize 9.0a. This is fixed by commenting out its removal (but not the other removal), like this:

https://github.com/danieldk/tgi-nix/blob/15bd4a978d6c2c8b04b7afc335d137dbe41e73df/pkgs/python-modules/torch/default.nix#L277

(Or the cmake derivation should be patched to modify the regexp to recognize the `a` suffix.)

--- a/pkgs/development/python-modules/torch/default.nix
+++ b/pkgs/development/python-modules/torch/default.nix
@@ -99,7 +99,7 @@ let
 
   setBool = v: if v then "1" else "0";
 
-  # https://github.com/pytorch/pytorch/blob/v2.0.1/torch/utils/cpp_extension.py#L1744
+  # https://github.com/pytorch/pytorch/blob/v2.4.0/torch/utils/cpp_extension.py#L1953
   supportedTorchCudaCapabilities =
     let
       real = [
@@ -119,6 +119,7 @@ let
         "8.7"
         "8.9"
         "9.0"
+        "9.0a"
       ];
       ptx = lists.map (x: "${x}+PTX") real;
     in
@@ -206,13 +207,14 @@ let
     # In particular, this triggered warnings from cuda's `aliases.nix`
     "Magma cudaPackages does not match cudaPackages" =
       cudaSupport && (effectiveMagma.cudaPackages.cudaVersion != cudaPackages.cudaVersion);
-    "Rocm support is currently broken because `rocmPackages.hipblaslt` is unpackaged. (2024-06-09)" = rocmSupport;
+    "Rocm support is currently broken because `rocmPackages.hipblaslt` is unpackaged. (2024-06-09)" =
+      rocmSupport;
   };
 in
 buildPythonPackage rec {
   pname = "torch";
   # Don't forget to update torch-bin to the same version.
-  version = "2.3.1";
+  version = "2.4.0";
   pyproject = true;
 
   disabled = pythonOlder "3.8.0";
@@ -230,11 +232,16 @@ buildPythonPackage rec {
     repo = "pytorch";
     rev = "refs/tags/v${version}";
     fetchSubmodules = true;
-    hash = "sha256-vpgtOqzIDKgRuqdT8lB/g6j+oMIH1RPxdbjtlzZFjV8=";
+    hash = "sha256-s49rtarGNNFpnNG+kfJtZLE8ND53Ma201I0cOjeFSts=";
   };
 
   patches =
-    lib.optionals cudaSupport [ ./fix-cmake-cuda-toolkit.patch ]
+    [
+      # Allow setting PYTHON_LIB_REL_PATH with an environment variable.
+      # https://github.com/pytorch/pytorch/pull/128419
+      ./passthrough-python-lib-rel-path.patch
+    ]
+    ++ lib.optionals cudaSupport [ ./fix-cmake-cuda-toolkit.patch ]
     ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
       # pthreadpool added support for Grand Central Dispatch in April
       # 2020. However, this relies on functionality (DISPATCH_APPLY_AUTO)
@@ -279,11 +286,12 @@ buildPythonPackage rec {
           'message(FATAL_ERROR "Found NCCL header version and library version' \
           'message(WARNING "Found NCCL header version and library version'
     ''
-    # Remove PyTorch's FindCUDAToolkit.cmake and to use CMake's default.
-    # We do not remove the entirety of cmake/Modules_CUDA_fix because we need FindCUDNN.cmake.
+    # Remove PyTorch's FindCUDAToolkit.cmake and use CMake's default.
+    # NOTE: Parts of pytorch rely on unmaintained FindCUDA.cmake with custom patches to support e.g.
+    # newer architectures (sm_90a). We do want to delete vendored patches, but have to keep them
+    # until https://github.com/pytorch/pytorch/issues/76082 is addressed
     + lib.optionalString cudaSupport ''
       rm cmake/Modules/FindCUDAToolkit.cmake
-      rm -rf cmake/Modules_CUDA_fix/{upstream,FindCUDA.cmake}
     ''
     # error: no member named 'aligned_alloc' in the global namespace; did you mean simply 'aligned_alloc'
     # This lib overrided aligned_alloc hence the error message. Tltr: his function is linkable but not in header.
@@ -374,6 +382,10 @@ buildPythonPackage rec {
   USE_NCCL = setBool useSystemNccl;
   USE_SYSTEM_NCCL = USE_NCCL;
   USE_STATIC_NCCL = USE_NCCL;
+
+  # Set the correct Python library path, broken since
+  # https://github.com/pytorch/pytorch/commit/3d617333e
+  PYTHON_LIB_REL_PATH = "${placeholder "out"}/${python.sitePackages}";
 
   # Suppress a weird warning in mkl-dnn, part of ideep in pytorch
   # (upstream seems to have fixed this in the wrong place?)
@@ -466,9 +478,7 @@ buildPythonPackage rec {
         libcusolver
         libcusparse
       ]
-      ++ lists.optionals (cudaPackages ? cudnn) [
-        cudnn
-      ]
+      ++ lists.optionals (cudaPackages ? cudnn) [ cudnn ]
       ++ lists.optionals useSystemNccl [
         # Some platforms do not support NCCL (i.e., Jetson)
         nccl # Provides nccl.h AND a static copy of NCCL!

--- a/pkgs/development/python-modules/torch/fix-cmake-cuda-toolkit.patch
+++ b/pkgs/development/python-modules/torch/fix-cmake-cuda-toolkit.patch
@@ -1,51 +1,15 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index 9194e520bb0..d05fdcfb6cb 100644
+index c4cd4b2c2a..e983b21353 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -1160,10 +1160,6 @@ if(BUILD_SHARED_LIBS)
-       ${PROJECT_SOURCE_DIR}/cmake/Modules_CUDA_fix
-       DESTINATION share/cmake/Caffe2/
-       COMPONENT dev)
--  install(FILES
--      ${PROJECT_SOURCE_DIR}/cmake/Modules/FindCUDAToolkit.cmake
--      DESTINATION share/cmake/Caffe2/
--      COMPONENT dev)
-   install(FILES
-       ${PROJECT_SOURCE_DIR}/cmake/Modules/FindCUSPARSELT.cmake
-       DESTINATION share/cmake/Caffe2/
-diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
-index c7595774d81..4fc43771810 100644
---- a/cmake/public/cuda.cmake
-+++ b/cmake/public/cuda.cmake
-@@ -61,9 +61,15 @@ find_package(CUDAToolkit REQUIRED)
- cmake_policy(POP)
- 
- if(NOT CMAKE_CUDA_COMPILER_VERSION VERSION_EQUAL CUDAToolkit_VERSION)
--  message(FATAL_ERROR "Found two conflicting CUDA versions:\n"
--                      "V${CMAKE_CUDA_COMPILER_VERSION} in '${CUDA_INCLUDE_DIRS}' and\n"
--                      "V${CUDAToolkit_VERSION} in '${CUDAToolkit_INCLUDE_DIRS}'")
-+  if(CUDA_INCLUDE_DIRS IN_LIST CUDAToolkit_INCLUDE_DIR)
-+    message(STATUS "CUDA_INCLUDE_DIRS is a substring of CUDAToolkit_INCLUDE_DIR. "
-+                   "Setting CUDA_INCLUDE_DIRS to CUDAToolkit_INCLUDE_DIR.")
-+    set(CUDA_INCLUDE_DIRS "${CUDAToolkit_INCLUDE_DIR}")
-+  else()
-+    message(FATAL_ERROR "Found two conflicting CUDA installs:\n"
-+                        "V${CMAKE_CUDA_COMPILER_VERSION} in '${CUDA_INCLUDE_DIRS}' and\n"
-+                        "V${CUDAToolkit_VERSION} in '${CUDAToolkit_INCLUDE_DIR}'")
-+  endif()
- endif()
- 
- if(NOT TARGET CUDA::nvToolsExt)
-diff --git a/tools/setup_helpers/cmake.py b/tools/setup_helpers/cmake.py
-index fb19b66dfba..3f83bef32fe 100644
---- a/tools/setup_helpers/cmake.py
-+++ b/tools/setup_helpers/cmake.py
-@@ -207,6 +207,8 @@ class CMake:
-                     "BUILDING_WITH_TORCH_LIBS",
-                     "CUDA_HOST_COMPILER",
-                     "CUDA_NVCC_EXECUTABLE",
-+                    "CUDAToolkit_ROOT",
-+                    "CUDAToolkit_INCLUDE_DIR",
-                     "CUDA_SEPARABLE_COMPILATION",
-                     "CUDNN_LIBRARY",
-                     "CUDNN_INCLUDE_DIR",
+@@ -1319,10 +1319,6 @@ if(BUILD_SHARED_LIBS)
+     DIRECTORY ${PROJECT_SOURCE_DIR}/cmake/Modules_CUDA_fix
+     DESTINATION share/cmake/Caffe2/
+     COMPONENT dev)
+-  install(
+-    FILES ${PROJECT_SOURCE_DIR}/cmake/Modules/FindCUDAToolkit.cmake
+-    DESTINATION share/cmake/Caffe2/
+-    COMPONENT dev)
+   install(
+     FILES ${PROJECT_SOURCE_DIR}/cmake/Modules/FindCUSPARSELT.cmake
+     DESTINATION share/cmake/Caffe2/

--- a/pkgs/development/python-modules/torch/passthrough-python-lib-rel-path.patch
+++ b/pkgs/development/python-modules/torch/passthrough-python-lib-rel-path.patch
@@ -1,0 +1,12 @@
+diff --git a/tools/setup_helpers/cmake.py b/tools/setup_helpers/cmake.py
+index 5481ce46031c..d50d9d547399 100644
+--- a/tools/setup_helpers/cmake.py
++++ b/tools/setup_helpers/cmake.py
+@@ -231,6 +231,7 @@ def generate(
+                     "SELECTED_OP_LIST",
+                     "TORCH_CUDA_ARCH_LIST",
+                     "TRACING_BASED",
++                    "PYTHON_LIB_REL_PATH",
+                 )
+             }
+         )

--- a/pkgs/development/python-modules/torchaudio/bin.nix
+++ b/pkgs/development/python-modules/torchaudio/bin.nix
@@ -16,7 +16,7 @@
 
 buildPythonPackage rec {
   pname = "torchaudio";
-  version = "2.3.1";
+  version = "2.4.0";
   format = "wheel";
 
   src =

--- a/pkgs/development/python-modules/torchaudio/binary-hashes.nix
+++ b/pkgs/development/python-modules/torchaudio/binary-hashes.nix
@@ -7,81 +7,81 @@
 
 version:
 builtins.getAttr version {
-  "2.3.1" = {
+  "2.4.0" = {
     x86_64-linux-38 = {
-      name = "torchaudio-2.3.1-cp38-cp38-linux_x86_64.whl";
-      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.3.1%2Bcu121-cp38-cp38-linux_x86_64.whl";
-      hash = "sha256-4QoP0bNR6EnnzZ4ekw3KP/uMow+WI5vqEx8YP2WUXNs=";
+      name = "torchaudio-2.4.0-cp38-cp38-linux_x86_64.whl";
+      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.4.0%2Bcu121-cp38-cp38-linux_x86_64.whl";
+      hash = "sha256-HbbFZ4nETaygQRxSMYg6d3Omqxbx6uSrXrUzRt71RTw=";
     };
     x86_64-linux-39 = {
-      name = "torchaudio-2.3.1-cp39-cp39-linux_x86_64.whl";
-      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.3.1%2Bcu121-cp39-cp39-linux_x86_64.whl";
-      hash = "sha256-bOv8CdkcpAPASLnPKuPWS09UzzGghYTQ0/v3+VAewNw=";
+      name = "torchaudio-2.4.0-cp39-cp39-linux_x86_64.whl";
+      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.4.0%2Bcu121-cp39-cp39-linux_x86_64.whl";
+      hash = "sha256-5wG1xXOR9ApLTdWQtt8eGt+83GwT6zfEW+Kay4pz5vQ=";
     };
     x86_64-linux-310 = {
-      name = "torchaudio-2.3.1-cp310-cp310-linux_x86_64.whl";
-      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.3.1%2Bcu121-cp310-cp310-linux_x86_64.whl";
-      hash = "sha256-C0I/SuM1bxH2cj6MdyCKw/k2Gk+UHkzAjYbDLBN1lLw=";
+      name = "torchaudio-2.4.0-cp310-cp310-linux_x86_64.whl";
+      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.4.0%2Bcu121-cp310-cp310-linux_x86_64.whl";
+      hash = "sha256-R+F2Pr3UEID6Otlv9OTyKHgxz6jd51IzrRMpxhOr2lA=";
     };
     x86_64-linux-311 = {
-      name = "torchaudio-2.3.1-cp311-cp311-linux_x86_64.whl";
-      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.3.1%2Bcu121-cp311-cp311-linux_x86_64.whl";
-      hash = "sha256-LZE2vzybarltxC8m9suDbUXRFfV2bhav9Sg1yOfN5WU=";
+      name = "torchaudio-2.4.0-cp311-cp311-linux_x86_64.whl";
+      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.4.0%2Bcu121-cp311-cp311-linux_x86_64.whl";
+      hash = "sha256-KcjEOkm0MDZnvyDNhcP26borSg+0tqrsLgKzstbMbOo=";
     };
     x86_64-linux-312 = {
-      name = "torchaudio-2.3.1-cp312-cp312-linux_x86_64.whl";
-      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.3.1%2Bcu121-cp312-cp312-linux_x86_64.whl";
-      hash = "sha256-sAdz3eOqMNmxuYh1ZQQo+/tp/jC5jrtc7/vY/GHegSU=";
+      name = "torchaudio-2.4.0-cp312-cp312-linux_x86_64.whl";
+      url = "https://download.pytorch.org/whl/cu121/torchaudio-2.4.0%2Bcu121-cp312-cp312-linux_x86_64.whl";
+      hash = "sha256-gTwbojelvYBZo0BOav9xtc09SXmtCV3PreFZJoCFUDE=";
     };
     aarch64-darwin-38 = {
-      name = "torchaudio-2.3.1-cp38-cp38-macosx_11_0_arm64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp38-cp38-macosx_11_0_arm64.whl";
-      hash = "sha256-zkXgWs1URpbGpvAj1P6GFK3ldRV5mhEDskGOhUg41KU=";
+      name = "torchaudio-2.4.0-cp38-cp38-macosx_11_0_arm64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp38-cp38-macosx_11_0_arm64.whl";
+      hash = "sha256-/D+OzW8Lv8ZU07xSdWp8o1nx2ItPoCkOHNt2OjExt7k=";
     };
     aarch64-darwin-39 = {
-      name = "torchaudio-2.3.1-cp39-cp39-macosx_11_0_arm64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp39-cp39-macosx_11_0_arm64.whl";
-      hash = "sha256-NujAtlMlccJ6CKQNrkKM00ryJQB/FbzXcnJkO2JmuB0=";
+      name = "torchaudio-2.4.0-cp39-cp39-macosx_11_0_arm64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp39-cp39-macosx_11_0_arm64.whl";
+      hash = "sha256-H9ZwyAjjIsEBlXoHZR4pk1+G7DiSQ8DEOiTt16GFSEE=";
     };
     aarch64-darwin-310 = {
-      name = "torchaudio-2.3.1-cp310-cp310-macosx_11_0_arm64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp310-cp310-macosx_11_0_arm64.whl";
-      hash = "sha256-H5E0sn5afwweMzgvwP4njlNpV2jLCvAujSK1AGx0oq0=";
+      name = "torchaudio-2.4.0-cp310-cp310-macosx_11_0_arm64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp310-cp310-macosx_11_0_arm64.whl";
+      hash = "sha256-cz6dhZuI2r7+rwCOOrK4x4hbKUZgaLS3mkJ2a+RhnkY=";
     };
     aarch64-darwin-311 = {
-      name = "torchaudio-2.3.1-cp311-cp311-macosx_11_0_arm64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp311-cp311-macosx_11_0_arm64.whl";
-      hash = "sha256-AZhPODmMpemOz7/q+3KuWyEx0LuKpGS1d3rds+SCaHc=";
+      name = "torchaudio-2.4.0-cp311-cp311-macosx_11_0_arm64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp311-cp311-macosx_11_0_arm64.whl";
+      hash = "sha256-rLz5Ep/8/OgIJU4sv/EDNjxQXOBu1MQjGz9DahBnnU0=";
     };
     aarch64-darwin-312 = {
-      name = "torchaudio-2.3.1-cp312-cp312-macosx_11_0_arm64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp312-cp312-macosx_11_0_arm64.whl";
-      hash = "sha256-XjZoVCCgehdhRunW4PqCJRmPEm4WegB4VTj4U4B+LUM=";
+      name = "torchaudio-2.4.0-cp312-cp312-macosx_11_0_arm64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp312-cp312-macosx_11_0_arm64.whl";
+      hash = "sha256-rhOpXvb6vK2w7/NthfUEjXBHSi6XBPqchumQPLzsDUo=";
     };
     aarch64-linux-38 = {
-      name = "torchaudio-2.3.1-cp38-cp38-manylinux2014_aarch64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp38-cp38-linux_aarch64.whl";
-      hash = "sha256-n9D0u8P9WF+9fZdqmI/m54P8suDbnXDaxg9AvgcsZQQ=";
+      name = "torchaudio-2.4.0-cp38-cp38-manylinux2014_aarch64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp38-cp38-linux_aarch64.whl";
+      hash = "sha256-1/6efy/oJQ/eB7IDVsRNdw1fqjyid6vc2jr31IQEj7o=";
     };
     aarch64-linux-39 = {
-      name = "torchaudio-2.3.1-cp39-cp39-manylinux2014_aarch64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp39-cp39-linux_aarch64.whl";
-      hash = "sha256-TjvKIy+CDGoPpTlEJAdsxRn64yKI5/9vbWi9cXlNw1Q=";
+      name = "torchaudio-2.4.0-cp39-cp39-manylinux2014_aarch64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp39-cp39-linux_aarch64.whl";
+      hash = "sha256-yECJTeEqbdPqV8uw0AhhI6qkgAG6Otme9xT+AJ6ujrk=";
     };
     aarch64-linux-310 = {
-      name = "torchaudio-2.3.1-cp310-cp310-manylinux2014_aarch64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp310-cp310-linux_aarch64.whl";
-      hash = "sha256-a1fnc6rXJ0PVCmSnQCoGy4vfzHCe/G2MJkKdlA5niOI=";
+      name = "torchaudio-2.4.0-cp310-cp310-manylinux2014_aarch64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp310-cp310-linux_aarch64.whl";
+      hash = "sha256-F8tz1DNncdRVzY3ai0iRMHpTRriQpOax1Lc9VlJY/uE=";
     };
     aarch64-linux-311 = {
-      name = "torchaudio-2.3.1-cp311-cp311-manylinux2014_aarch64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp311-cp311-linux_aarch64.whl";
-      hash = "sha256-yMcnyDQYJb0Y2RAXxMAPNrU7CPIXbNub3LDe8cRQsh0=";
+      name = "torchaudio-2.4.0-cp311-cp311-manylinux2014_aarch64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp311-cp311-linux_aarch64.whl";
+      hash = "sha256-vpacCUZts14Nebiwnf9myu27lWm0LJA6LV4Nsq92Djw=";
     };
     aarch64-linux-312 = {
-      name = "torchaudio-2.3.1-cp312-cp312-manylinux2014_aarch64.whl";
-      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.3.1-cp312-cp312-linux_aarch64.whl";
-      hash = "sha256-Qq9sekMOYmjywCjgYHjUE5ErXsbvoooJfr3Tw8eWWd8=";
+      name = "torchaudio-2.4.0-cp312-cp312-manylinux2014_aarch64.whl";
+      url = "https://download.pytorch.org/whl/cpu/torchaudio-2.4.0-cp312-cp312-linux_aarch64.whl";
+      hash = "sha256-U00ZB7slLs0rqeHWHP9yIP1mCQ5j33s8EJzqd6GdTLg=";
     };
   };
 }

--- a/pkgs/development/python-modules/torchvision/default.nix
+++ b/pkgs/development/python-modules/torchvision/default.nix
@@ -18,7 +18,7 @@ let
   inherit (cudaPackages) backendStdenv;
 
   pname = "torchvision";
-  version = "0.18.1";
+  version = "0.19.0";
 in
 buildPythonPackage {
   inherit pname version;
@@ -27,7 +27,7 @@ buildPythonPackage {
     owner = "pytorch";
     repo = "vision";
     rev = "refs/tags/v${version}";
-    hash = "sha256-aFm6CyoMA8HtpOAVF5Q35n3JRaOXYswWEqfooORUKsw=";
+    hash = "sha256-OVwdEqKEoZR1jtcg4ODyvIZvP9UQPodPv8qnTqbA/pc=";
   };
 
   nativeBuildInputs = [
@@ -76,6 +76,7 @@ buildPythonPackage {
   meta = {
     description = "PyTorch vision library";
     homepage = "https://pytorch.org/";
+    changelog = "https://github.com/pytorch/vision/releases/tag/v${version}";
     license = lib.licenses.bsd3;
     platforms = with lib.platforms; linux ++ lib.optionals (!cudaSupport) darwin;
     maintainers = with lib.maintainers; [ ericsagnes ];

